src/Hal/JsonSchema/SchemaFactory.php

```diff
@@ -79,8 +78,21 @@ public function buildSchema(string $className, string $format = 'jsonhal', strin
             $schema['type'] = 'object';
             $schema['properties'] = [
                 '_embedded' => [
-                    'type' => 'array',
-                    'items' => $items,
+                    'anyOf' => [
+                        [
+                            'type' => 'object',
+                            'properties' => [
+                                'items' => [
+                                    'type' => 'array',
+                                    'items' => $items,
+                                ],
+                            ],
+                        ],
+                        [
+                            'type' => 'array',
+                            'items' => $items,
+                        ],
+                    ],
```

Comment:
I'm not 100% sure if this should be an `anyOf`, or just the `object` … Plus shouldn't `items` the the resource name?

Reply:
The `items` thing comes from: https://json-schema.org/understanding-json-schema/reference/array#items

But in `hal` `_embedded` is always an object right?

Reply:
Ah, good to know! Thanks you.

I believe it is, I just looked at `ApiPlatform\Hal\Serializer\CollectionNormalizer::getItemsData()` and it is, I'll adjust the above change.